cdap-cli/src/main/java/co/cask/cdap/cli/util/ArgumentParser.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.util;
+
+import co.cask.common.cli.util.Parser;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static co.cask.common.cli.util.Parser.MANDATORY_ARG_BEGINNING;
+import static co.cask.common.cli.util.Parser.MANDATORY_ARG_ENDING;
+import static co.cask.common.cli.util.Parser.OPTIONAL_PART_BEGINNING;
+import static co.cask.common.cli.util.Parser.OPTIONAL_PART_ENDING;
+
+/**
+ * Utility class for parsing arguments from user input based on command pattern.
+ */
+public class ArgumentParser {
+
+  /**
+   * Parses program id from user input.
+   * For example, HelloWorld.Greeting will be parsed to [HelloWorld, Greeting].
+   *
+   * @param programId the program id
+   * @return split id
+   */
+  public static String[] parseProgramId(String programId) {
```

Comment:
Should return some sort of object from here like `ProgramIdArgument` which contains appId and programId.